django/db/models/sql/compiler.py

```diff
@@ -605,21 +610,22 @@ def get_distinct(self):
         This method can alter the tables in the query, and thus it must be
         called before get_from_clause().
         """
-        qn = self.quote_name_unless_alias
-        qn2 = self.connection.ops.quote_name
         result = []
+        params = []
         opts = self.query.get_meta()
 
         for name in self.query.distinct_fields:
             parts = name.split(LOOKUP_SEP)
-            _, targets, alias, joins, path, _ = self._setup_joins(parts, opts, None)
+            field, targets, alias, joins, path, _, transform_function = self._setup_joins(parts, opts, None)
             targets, alias, _ = self.query.trim_joins(targets, joins, path)
             for target in targets:
                 if name in self.query.annotation_select:
                     result.append(name)
                 else:
-                    result.append("%s.%s" % (qn(alias), qn2(target.column)))
-        return result
+                    r, p = transform_function(target, alias).as_sql(self, self.connection)
```

Comment:
I'm not sure how this would change to do that, does `as_sql` not take the backend into account? I can't find reference to `as_backend()` in the codebase I'm afraid.

Reply:
`.as_sql()` is the default implementation. `as_{backend.vendor}` is the backend specific implementation *if it exists*.

`compiler.compile` checks if a backend specific implementation exists first and uses that, falling back to `as_sql` if one does not exist.

So, `self.compile(transform_function(..))` I think would be correct.

Reply:
Ah, thanks! I'll make this change.